app/domain/payment_provider.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2021, Die Mitte Schweiz. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class PaymentProvider
+  def initialize(config)
+    @config = config
+  end
+
+  def initial_setup
+    @client = Epics::Client.setup(config.password,
+                                  payment_provider_setting.url,
+                                  payment_provider_setting.host_id,
+                                  config.user_identifier,
+                                  config.partner_identifier)
+
+    config.update(keys: client.send(:dump_keys))
+  end
+
+  def INI
+    ok = client.INI
+
+    raise 'INI request failed' unless ok
+  end
+
+  def ini_letter
+    @ini_letter ||= client.ini_letter(config.payment_provider)
+  end
+
+  def HIA
+    ok = client.HIA
+
+    raise 'HIA request failed' unless ok
+  end
+
+  def HPB
+    bank_x, bank_e = client.HPB
+
+    check_bank_public_keys!(bank_x, bank_e)
+  end
+
+  private
+
+  attr_reader :config
```

Comment:
Ich habe den Sinn von `private attr_reader` noch nie verstanden. Der Effekt ist klar, nur macht es aus meiner Sicht den Code schwerer zu lesen. Man liest `config` und muss erstmal suchen, wo das herkommt. Bei `@config` ist es IMHO klarer. 

Und ja, der Nachteil ist natürlich, dass man mehr anpassen muss, wenn man das nachträglich ändern möchte. 

Mal unabhängig von Standards und Konventionen: was liest sich für dich besser? Instanzvariable oder Methodenaufruf?

Reply:
Hmm hab mir das noch gar nie so genau überlegt. In diesem Fall wo die Variable allein im Konstruktor gesetzt und danach nur noch ausgelesen wird, ist wohl `@config` das klar lesbarste 🤔